src/EnsembleKalmanInversion.jl

```diff
@@ -77,8 +76,8 @@ end
         u::AbstractMatrix{FT},
         g::AbstractMatrix{FT},
         y::AbstractMatrix{FT},
-        obs_noise_cov::Union{AbstractMatrix{FT}, UniformScaling{FT}},
-    ) where {FT <: Real, IT}
+        obs_noise_cov::Union{AbstractMatrix{CT}, UniformScaling{CT}},
```

Comment:
I think maybe `obs_noise_cov` should stay linked? Was there a case you had in mind for when they differ?

Just looking ahead we do `(cov_gg + obs_noise_cov) \ (y - g)` and presumably we don't want any `FT` vs `CT` type issues here.

Reply:
I was thinking of this case:

```
julia> using Random

julia> using LinearAlgebra

julia> cov1 = I
UniformScaling{Bool}
true*I

julia> cov2 = 1.0 * I
UniformScaling{Float64}
1.0*I

julia> y = rand(5)
5-element Vector{Float64}:
 0.8876949829922091
 0.9351539077673359
 0.992727497437651
 0.6203034498205577
 0.9959225311210873

julia> g = rand(5)
5-element Vector{Float64}:
 0.9247820413465523
 0.6061500089264417
 0.739520036070634
 0.3345260433372921
 0.606807569247004

julia> res1 = cov1 \ (y - g)
5-element Vector{Float64}:
 -0.03708705835434323
  0.3290038988408942
  0.253207461367017
  0.28577740648326566
  0.38911496187408334

julia> res2 = cov2 \ (y - g)
5-element Vector{Float64}:
 -0.03708705835434323
  0.3290038988408942
  0.253207461367017
  0.28577740648326566
  0.38911496187408334

julia> res1 == res2
true
```

What I can do is cast the result to FT to ensure type stable computation, but I prefer to not be so strict about input typing.